### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by aepaysinger**
not sure why the test isnt working 
platform linux -- Python 3.10.4, pytest-7.4.0, pluggy-1.3.0
rootdir: /home/circleci/project
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
_____________________ ERROR collecting test_linkedlist.py ______________________
ImportError while importing test module '/home/circleci/project/test_linkedlist.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../.pyenv/versions/3.10.4/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_linkedlist.py:3: in <module>
    from linkedlist import LinkedList, Node
E   ModuleNotFoundError: No module named 'linkedlist'
=========================== short test summary info ============================
ERROR test_linkedlist.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.08s ===============================

Exited with code exit status 2
it is saying no module name linkedlist, but it is there so im confused

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # See: https://circleci.com/docs/configuration-reference/#executor-job
+    docker:
+      - image: cimg/python:3.10.4
+    # Add steps to the job
+    # See: https://circleci.com/docs/configuration-reference/#steps
+    steps:
+      - checkout
+      - run:
+          name: "install requirements"
+          command: "pip install -r requirements.txt"
```
Comment (W-Ely) [inline .circleci/config.yml]:
```suggestion
          command: |
            python3 -m venv venv
            . venv/bin/activate
            pip install -r requirements.txt
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - run:
+          name: "run tests"
+          command: "pytest"
```
Comment (W-Ely) [inline .circleci/config.yml]:
```suggestion
          command: |
            . venv/bin/activate
            pytest
```

**Review by W-Ely — COMMENTED** [review]
Might try these.

File: .circleci/config.yml (hunk 2 above)
Comment (aepaysinger) [reply .circleci/config.yml]: i tried the changes but in visual studio it had a bunch of red

File: .circleci/config.yml [hunk 3]
```diff
@@ -29,10 +29,13 @@ jobs:
...
-      - run:
-          name: "run pytest"
-          command: "pytest"
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+          command: |
+            . venv/bin/activate
+            pytest
```
Comment (W-Ely) [inline .circleci/config.yml]: This is not correct!  I made 2 separate suggestions, you can't combine them like this.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      - checkout
+      - run:
+          name: "install requirements"
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+      - run:
+          name: "run tests"
+          command: "pytest"
```
Comment (W-Ely) [inline .circleci/config.yml]:
```suggestion
          command: |
            . venv/bin/activate
            pytest
```

File: tests/test_linkedlist.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import io, pytest, sys
```
Comment (W-Ely) [inline tests/test_linkedlist.py]: You are changing to many things at the same time.  But now that you have changed the directory structure, you need to add an `__init__.py` file in the test folder too.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+          name: "install requirements"
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+      - run:
+          name: "run tests"
+          command: |
+            . venv/bin/activate
+            pytest
```
Comment (W-Ely) [inline .circleci/config.yml]: Might try this:

```suggestion
            PYTHONPATH=. pytest
```

**Review by W-Ely — APPROVED** [review]

**Pull request merged**